### PR TITLE
chore: Add missing config values to mapper

### LINKED ISSUE #285: chore: Add all newer configuration items to Environment Variable mapping.
## What happened
Currently, there is some missing support for some configurable values via environment variables.

## What should to happen
In order to support environment variables for configuration classes, we need to map the conventional screaming snake case for environment variables (since still by default we are not able to get another case at runtime) to a convention the config api understands(e.g. `MEDIATOR_RING_BUFFER_SIZE` to `mediator.ringBufferSize`). This is done by updating the `com.hedera.block.server.config.ServerMappedConfigSourceInitializer#MAPPINGS` and we also need to update the respective awareness test `com.hedera.block.server.config.ServerMappedConfigSourceInitializerTest#SUPPORTED_MAPPINGS`. Also, alongside that, we should update the `README.md` to let the users know what are the correct screaming snake case keys for the configs.

## Tasklist
- [ ] Update all the other needed configurations that need to be supported to be set via environment variables in `:server`
- [ ] Update all the other needed configurations that need to be supported to be set via environment variables in `:simulator`

Missing configuration that should be added:
- From `server`:
   - producer.type → PRODUCER_TYPE (default: "PRODUCTION")
   - mediator.type → MEDIATOR_TYPE (default: "PRODUCTION")
- From `simulator`:
   - From GrpcConfig.java:
      - grpc.serverAddress (default: "localhost")
      - grpc.port (default: 8080)
   - From BlockStreamConfig.java:
      - blockStream.simulatorMode (default: "PUBLISHER")
      - blockStream.delayBetweenBlockItems (default: 1_500_000)
      - blockStream.maxBlockItemsToStream (default: 100_000)
      - blockStream.streamingMode (default: "MILLIS_PER_BLOCK")
      - blockStream.millisecondsPerBlock (default: 1000)
      - blockStream.blockItemsBatchSize (default: 1000)
   - From BlockGeneratorConfig.java:
      - generator.generationMode (default: "DIR")
      - generator.folderRootPath (default: "")
      - generator.managerImplementation (default: "BlockAsFileBlockStreamManager")
      - generator.paddedLength (default: 36)
      - generator.fileExtension (default: ".blk.gz")
      - generator.startBlockNumber (default: 1)
      - generator.endBlockNumber (default: -1)
## Version

0.2.0-SNAPSHOT

### PR DESCRIPTION
**Pull request by georgi-l95**
**Description**:
This PR makes the following changes:
- On the `server` side adds the following missing configs to the mapper:
   - producer.type → PRODUCER_TYPE (default: "PRODUCTION")
   - mediator.type → MEDIATOR_TYPE (default: "PRODUCTION")
- On the `simulator` side adds the following missing configs to the mapper:
   - From GrpcConfig.java:
      - grpc.serverAddress (default: "localhost")
      - grpc.port (default: 8080)
   - From BlockStreamConfig.java:
      - blockStream.simulatorMode (default: "PUBLISHER")
      - blockStream.delayBetweenBlockItems (default: 1_500_000)
      - blockStream.maxBlockItemsToStream (default: 100_000)
      - blockStream.streamingMode (default: "MILLIS_PER_BLOCK")
      - blockStream.millisecondsPerBlock (default: 1000)
      - blockStream.blockItemsBatchSize (default: 1000)
   - From BlockGeneratorConfig.java:
      - generator.generationMode (default: "DIR")
      - generator.folderRootPath (default: "")
      - generator.managerImplementation (default: "BlockAsFileBlockStreamManager")
      - generator.paddedLength (default: 36)
      - generator.fileExtension (default: ".blk.gz")
      - generator.startBlockNumber (default: 1)
      - generator.endBlockNumber (default: -1)

Additional external variables added to both are:
- prometheus.endpointEnabled (default: true)
- prometheus.endpointPortNumber (default: 9999)

Adds two new gradle tasks for the simulator: 
- runPublisher (starts simulator in Publisher mode with enabled prometheus on port 9998)
- runConsumer (starts simulator in Consumer mode with enabled prometheus on port 9997)

**Related issue(s)**:

Fixes #285 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: simulator/build.gradle.kts [hunk 1]
```diff
@@ -44,6 +44,34 @@ testModuleInfo {
...
+    group = "application"
+
+    mainClass = application.mainClass
+    mainModule = application.mainModule
+    classpath = sourceSets["main"].runtimeClasspath
+
+    environment("BLOCK_STREAM_SIMULATOR_MODE", "CONSUMER")
+    environment("PROMETHEUS_ENDPOINT_ENABLED", "true")
+    environment("PROMETHEUS_ENDPOINT_PORT_NUMBER", "9997")
+}
```
Comment (mattp-swirldslabs) [inline simulator/build.gradle.kts:73]: I like these new gradle tasks! I don't have to fuss around with changing the `app.properties` files (and then remember to revert the changes).

**Review by mattp-swirldslabs — APPROVED** [review]
Nice work. LGTM.

File: simulator/build.gradle.kts (hunk 1 above)
Comment (AlfredoG87) [reply simulator/build.gradle.kts:73]: agreed! 👍

**Review by AlfredoG87 — APPROVED** [review]
LGTM
